modules/govuk_scripts/templates/govuk_node_list_aws.erb

```diff
@@ -55,21 +55,21 @@ def ec2_nodes_list_with_puppet_class(stackname):
                )
 
     reservations = response['Reservations']
-    for reservation in reservations:
-        instances = reservation['Instances']
 
-    return [
-        "%s:%s" % (
-            instance['PrivateDnsName'],
-            next(
-                filter(
-                    lambda t: t['Key'] == 'aws_hostname',
-                    instance['Tags']
-                )
-            )['Value']
-        )
-        for instance in list(instances)
-    ]
+    hostname_and_puppet_class = []
+
+    for reservation in reservations:
+        for instance in list(reservation['Instances']):
+            hostname_and_puppet_class.append("%s:%s" % (
```

Comment:
It might be nicer to use format strings here, something like:

```python
private_dns_name = instance['PrivateDnsName']
hostname = ...
hostname_and_puppet_class.append(f"{private_dns_name}:{hostname}")
```

Reply:
Oooh, yes!

Reply:
We don't have a new enough version of Python (we've got 3.4.3 on the integration jumpbox where I'm doing testing) for that, so we've gone with:

`"{private_dns_name}:{hostname}".format(private_dns_name=private_dns_name, hostname=hostname)`